diff --git a/subprojects/gst-plugins-good/ext/adaptivedemux2/dash/gstdashdemux.c b/subprojects/gst-plugins-good/ext/adaptivedemux2/dash/gstdashdemux.c
index ab7445f532..b2a8a801e9 100644
--- a/subprojects/gst-plugins-good/ext/adaptivedemux2/dash/gstdashdemux.c
+++ b/subprojects/gst-plugins-good/ext/adaptivedemux2/dash/gstdashdemux.c
@@ -1634,7 +1634,7 @@ gst_dash_demux_stream_has_next_subfragment (GstAdaptiveDemux2Stream * stream)
   if (dashstream->sidx_parser.status == GST_ISOFF_SIDX_PARSER_FINISHED) {
     gboolean playing_forward = (stream->demux->segment.rate > 0.0);
     if (playing_forward) {
-      if (sidx->entry_index + 1 < sidx->entries_count)
+      if (sidx->entry_index < sidx->entries_count)
         return TRUE;
     } else {
       if (sidx->entry_index >= 1)