diff --git a/libsecret/meson.build b/libsecret/meson.build
index 3c534d0..07830b1 100644
--- a/libsecret/meson.build
+++ b/libsecret/meson.build
@@ -36,13 +36,13 @@ libsecret_headers = [
   'secret-value.h',
 ]
 
-if with_crypto
+# if with_crypto
   libsecret_sources += [
     'secret-file-backend.c',
     'secret-file-collection.c',
     'secret-file-item.c',
   ]
-endif
+# endif
 
 version_numbers = meson.project_version().split('.')
 version_major = version_numbers[0].to_int()
@@ -101,6 +101,7 @@ libsecret_dep = declare_dependency(
   link_with: [ libsecret, libegg ],
   dependencies: libsecret_dependencies,
+  include_directories: config_h_dir,
 )
 
 # GObject Introspection

@@ -231,11 +231,11 @@ test_names = [
   'test-collection',
 ]
 
-if with_crypto
+# if with_crypto
   test_names += [
     'test-file-collection',
   ]
-endif
+# endif
 
 foreach _test : test_names
 
diff --git a/libsecret/secret-backend.c b/libsecret/secret-backend.c
index 75c42da..cdd05ca 100644
--- a/libsecret/secret-backend.c
+++ b/libsecret/secret-backend.c
@@ -16,9 +16,9 @@
 
 #include "secret-backend.h"
 
-#ifdef WITH_CRYPTO
+// #ifdef WITH_CRYPTO
 #include "secret-file-backend.h"
-#endif
+// #endif
 
 #include "secret-private.h"
 
@@ -148,16 +148,16 @@ backend_get_impl_type (void)
 	GIOExtensionPoint *ep;
 
 	g_type_ensure (secret_service_get_type ());
-#ifdef WITH_CRYPTO
+// #ifdef WITH_CRYPTO
 	g_type_ensure (secret_file_backend_get_type ());
-#endif
+// #endif
 
-#ifdef WITH_CRYPTO
+// #ifdef WITH_CRYPTO
 	if ((g_file_test ("/.flatpak-info", G_FILE_TEST_EXISTS) || g_getenv ("SNAP_NAME") != NULL) &&
 	    _secret_file_backend_check_portal_version ())
 		extension_name = "file";
 	else
-#endif
+// #endif
 	{
 		envvar = g_getenv ("SECRET_BACKEND");
 		if (envvar == NULL || *envvar == '\0')
diff --git a/libsecret/secret-file-collection.c b/libsecret/secret-file-collection.c
index b33caad..b78837c 100644
--- a/libsecret/secret-file-collection.c
+++ b/libsecret/secret-file-collection.c
@@ -16,14 +16,16 @@
 
 #include "secret-file-collection.h"
 
-#include "egg/egg-keyring1.h"
 #include "egg/egg-secure-memory.h"
 
 EGG_SECURE_DECLARE (secret_file_collection);
 
+#ifdef WITH_CRYPTO
+#include "egg/egg-keyring1.h"
 #ifdef WITH_GCRYPT
 #include "egg/egg-libgcrypt.h"
 #endif
+#endif
 
 #define KEYRING_FILE_HEADER "GnomeKeyring\n\r\0\n"
 #define KEYRING_FILE_HEADER_LEN 16
@@ -166,8 +168,10 @@ load_contents (SecretFileCollection *self,
 	guint64 usage_count;
 	gconstpointer data;
 	gsize n_data;
+#ifdef WITH_CRYPTO
 	const gchar *password;
 	gsize n_password;
+#endif
 
 	p = contents;
 	if (length < KEYRING_FILE_HEADER_LEN ||
@@ -219,6 +223,7 @@ load_contents (SecretFileCollection *self,
 	g_variant_unref (salt_array);
 	g_variant_unref (variant);
 
+#ifdef WITH_CRYPTO
 	password = secret_value_get (self->password, &n_password);
 	self->key = egg_keyring1_derive_key (password,
 					     n_password,
@@ -231,6 +236,7 @@ load_contents (SecretFileCollection *self,
 				     "couldn't derive key");
 		return FALSE;
 	}
+#endif
 
 	return TRUE;
 }
@@ -240,6 +246,7 @@ init_empty_file (SecretFileCollection *self,
 		 GError **error)
 {
 	GVariantBuilder builder;
+#ifdef WITH_CRYPTO
 	const gchar *password;
 	gsize n_password;
 	guint8 salt[SALT_SIZE];
@@ -247,9 +254,14 @@ init_empty_file (SecretFileCollection *self,
 	egg_keyring1_create_nonce (salt, sizeof(salt));
 	self->salt = g_bytes_new (salt, sizeof(salt));
 	self->iteration_count = ITERATION_COUNT;
+#else
+	self->salt = g_bytes_new (NULL, 0);
+	self->iteration_count = 0;
+#endif
 	self->modified = g_date_time_new_now_utc ();
 	self->usage_count = 0;
 
+#ifdef WITH_CRYPTO
 	password = secret_value_get (self->password, &n_password);
 	self->key = egg_keyring1_derive_key (password,
 					     n_password,
@@ -262,6 +274,7 @@ init_empty_file (SecretFileCollection *self,
 				     "couldn't derive key");
 		return FALSE;
 	}
+#endif
 
 	g_variant_builder_init (&builder,
 				G_VARIANT_TYPE ("a(a{say}ay)"));
@@ -393,7 +406,9 @@ hash_attributes (SecretFileCollection *self,
 		 GHashTable *attributes)
 {
 	GVariantBuilder builder;
+#ifdef WITH_CRYPTO
 	guint8 buffer[MAC_SIZE];
+#endif
 	GList *keys;
 	GList *l;
 
@@ -407,6 +422,7 @@ hash_attributes (SecretFileCollection *self,
 		GVariant *variant;
 
 		value = g_hash_table_lookup (attributes, l->data);
+#ifdef WITH_CRYPTO
 		if (!egg_keyring1_calculate_mac (self->key,
 						 (const guint8 *)value,
 						 strlen (value),
@@ -419,6 +435,12 @@ hash_attributes (SecretFileCollection *self,
 						     buffer,
 						     MAC_SIZE,
 						     sizeof(guint8));
+#else
+		variant = g_variant_new_fixed_array (G_VARIANT_TYPE_BYTE,
+						     value,
+						     strlen (value),
+						     sizeof(guint8));
+#endif
 		g_variant_builder_add (&builder, "{s@ay}", l->data, variant);
 	}
 	g_list_free (keys);
@@ -447,6 +469,7 @@ hashed_attributes_match (SecretFileCollection *self,
 
 		data = g_variant_get_fixed_array (hashed_attribute,
 						  &n_data, sizeof(guint8));
+#ifdef WITH_CRYPTO
 		if (n_data != MAC_SIZE) {
 			g_variant_unref (hashed_attribute);
 			return FALSE;
@@ -456,6 +479,13 @@ hashed_attributes_match (SecretFileCollection *self,
 			g_variant_unref (hashed_attribute);
 			return FALSE;
 		}
+#else
+		if (n_data != strlen ((char *)value) ||
+			memcmp (data, value, n_data) != 0) {
+			g_variant_unref (hashed_attribute);
+			return FALSE;
+		}
+#endif
 		g_variant_unref (hashed_attribute);
 	}
 
@@ -477,7 +507,9 @@ secret_file_collection_replace (SecretFileCollection *self,
 	GVariant *serialized_item;
 	guint8 *data = NULL;
 	gsize n_data;
+#ifdef WITH_CRYPTO
 	gsize n_padded;
+#endif
 	GVariant *variant;
 	GDateTime *created = NULL;
 	GDateTime *modified;
@@ -540,6 +572,7 @@ secret_file_collection_replace (SecretFileCollection *self,
 	serialized_item = secret_file_item_serialize (item);
 	g_object_unref (item);
 
+#ifdef WITH_CRYPTO
 	/* Encrypt the item with PKCS #7 padding */
 	n_data = g_variant_get_size (serialized_item);
 	n_padded = ((n_data + CIPHER_BLOCK_SIZE) / CIPHER_BLOCK_SIZE) *
@@ -577,6 +610,23 @@ secret_file_collection_replace (SecretFileCollection *self,
 					   TRUE,
 					   egg_secure_free,
 					   data);
+#else
+	n_data = g_variant_get_size (serialized_item);
+	data = egg_secure_alloc (n_data);
+	g_variant_store (serialized_item, data);
+	g_variant_unref (serialized_item);
+
+	self->usage_count++;
+	g_date_time_unref (self->modified);
+	self->modified = g_date_time_new_now_utc ();
+
+	variant = g_variant_new_from_data (G_VARIANT_TYPE ("ay"),
+					   data,
+					   n_data,
+					   TRUE,
+					   egg_secure_free,
+					   data);
+#endif
 	variant = g_variant_new ("(@a{say}@ay)", hashed_attributes, variant);
 	g_variant_builder_add_value (&builder, variant);
 
@@ -630,12 +680,13 @@ _secret_file_item_decrypt (GVariant *encrypted,
 
 	g_variant_get (encrypted, "(a{say}@ay)", NULL, &blob);
 
-	/* Decrypt the item */
 	padded = g_variant_get_fixed_array (blob, &n_padded, sizeof(guint8));
 	data = egg_secure_alloc (n_padded);
 	memcpy (data, padded, n_padded);
 	g_variant_unref (blob);
 
+#ifdef WITH_CRYPTO
+	/* Decrypt the item */
 	if (n_padded < IV_SIZE + MAC_SIZE) {
 		egg_secure_free (data);
 		g_set_error (error,
@@ -667,6 +718,9 @@ _secret_file_item_decrypt (GVariant *encrypted,
 
 	/* Remove PKCS #7 padding */
 	n_data = n_padded - data[n_padded - 1];
+#else
+	n_data = n_padded;
+#endif
 
 	serialized_item =
 		g_variant_new_from_data (G_VARIANT_TYPE ("(a{ss}sttay)"),
diff --git a/libsecret/test-file-collection.c b/libsecret/test-file-collection.c
index 512bfb6..ee75c43 100644
--- a/libsecret/test-file-collection.c
+++ b/libsecret/test-file-collection.c
@@ -308,6 +308,7 @@ test_write (Test *test,
 	g_main_loop_run (test->loop);
 }
 
+#ifdef WITH_CRYPTO
 static void
 test_read (Test *test,
 	   gconstpointer unused)
@@ -350,6 +351,7 @@ test_read (Test *test,
 	g_object_unref (item);
 	g_hash_table_unref (attributes);
 }
+#endif
 
 int
 main (int argc, char **argv)
@@ -362,7 +364,9 @@ main (int argc, char **argv)
 	g_test_add ("/file-collection/search", Test, NULL, setup, test_search, teardown);
 	g_test_add ("/file-collection/decrypt", Test, NULL, setup, test_decrypt, teardown);
 	g_test_add ("/file-collection/write", Test, NULL, setup, test_write, teardown);
+#ifdef WITH_CRYPTO
 	g_test_add ("/file-collection/read", Test, "default.keyring", setup, test_read, teardown);
+#endif
 
 	return egg_tests_run_with_loop ();
 }
