diff --git a/tool/secret-tool.c b/tool/secret-tool.c
index c72e403..829c44c 100644
--- a/tool/secret-tool.c
+++ b/tool/secret-tool.c
@@ -281,15 +281,15 @@ read_password_stdin (void)
 	}
 }
 
-static SecretValue *
-read_password_tty (void)
-{
-	gchar *password;
+// static SecretValue *
+// read_password_tty (void)
+// {
+// 	gchar *password;
 
-	password = getpass ("Password: ");
-	return secret_value_new_full (password, -1, "text/plain",
-	                              (GDestroyNotify)secret_password_wipe);
-}
+// 	password = getpass ("Password: ");
+// 	return secret_value_new_full (password, -1, "text/plain",
+// 	                              (GDestroyNotify)secret_password_wipe);
+// }
 
 static int
 secret_tool_action_store (int argc,
@@ -327,10 +327,7 @@ secret_tool_action_store (int argc,
 			collection = g_strconcat (SECRET_ALIAS_PREFIX, store_collection, NULL);
 	}
 
-	if (isatty (0))
-		value = read_password_tty ();
-	else
-		value = read_password_stdin ();
+	value = read_password_stdin ();
 
 	ret = secret_password_storev_binary_sync (NULL, attributes, collection, store_label, value, NULL, &error);
 	secret_value_unref (value);
